pkg/controller/controller.go

```diff
@@ -47,55 +52,95 @@ var knownKinds = []knownKind{{
 
 // Workload represents a workload in the cluster. It contains the top level object and all of the pods.
 type Workload struct {
-	TopController unstructured.Unstructured
-	Pods          []unstructured.Unstructured
+	TopController   unstructured.Unstructured
+	Pods            []unstructured.Unstructured
+	PodSpec         *corev1.PodSpec
+	PodCount        int
+	RunningPodCount int
 }
 
-func getAllPods(ctx context.Context, dynamicClient dynamic.Interface, restMapper meta.RESTMapper, namespace string) ([]unstructured.Unstructured, error) {
+// Client is used to interact with the Kubernetes API
+type Client struct {
+	Context    context.Context
+	Dynamic    dynamic.Interface
+	RESTMapper meta.RESTMapper
+}
+
+func (client Client) getAllPods(namespace string) ([]unstructured.Unstructured, error) {
 	fqKind := schema.FromAPIVersionAndKind("v1", "Pod")
-	mapping, err := restMapper.RESTMapping(fqKind.GroupKind(), fqKind.Version)
+	mapping, err := client.RESTMapper.RESTMapping(fqKind.GroupKind(), fqKind.Version)
 	if err != nil {
 		log.GetLogger().Error(err, "Error retrieving mapping", "v1", "Pod")
 		return nil, err
 	}
-	pods, err := dynamicClient.Resource(mapping.Resource).Namespace(namespace).List(ctx, metav1.ListOptions{})
+	pods, err := client.Dynamic.Resource(mapping.Resource).Namespace(namespace).List(client.Context, metav1.ListOptions{})
 	if err != nil {
 		return nil, err
 	}
 	return pods.Items, nil
 }
 
-func prepCacheWithKnownControllers(ctx context.Context, dynamicClient dynamic.Interface, restMapper meta.RESTMapper, namespace string, objectCache map[string]unstructured.Unstructured) error {
+func getPodStatus(unst unstructured.Unstructured) string {
+	obj := unst.UnstructuredContent()
+	if statusI, ok := obj["status"]; ok {
+		if status, ok := statusI.(map[string]interface{}); ok {
+			if phaseI, ok := status["phase"]; ok {
+				if phase, ok := phaseI.(string); ok {
+					return phase
+				}
+			}
+		}
+	}
+	return ""
+}
+
+func (client Client) prepCacheWithKnownControllers(namespace string, objectCache map[string]unstructured.Unstructured) error {
 	for _, kind := range knownKinds {
-		err := cacheAllObjectsOfKind(ctx, kind.apiVersion, kind.kind, namespace, dynamicClient, restMapper, objectCache)
+		err := client.cacheAllObjectsOfKind(kind.apiVersion, kind.kind, namespace, objectCache, true)
 		if err != nil {
 			log.GetLogger().V(3).Info("Unable to prime cache with objects of kind " + kind.kind)
 		}
 	}
 	return nil
 }
 
-// GetAllTopControllers returns the highest level owning object of all pods. If a namespace is provided than this is limited to that namespace.
-func GetAllTopControllers(ctx context.Context, dynamicClient dynamic.Interface, restMapper meta.RESTMapper, namespace string) ([]Workload, error) {
+// GetAllTopControllersSummary returns the highest level owning object of all pods, as well as all pods.
```

Comment:
Is this comment correct? My understanding is that it _does not_ return pods (at least according to the final sentence in this string

Reply:
Nope :facepalm: copypast